### ISSUE TITLE
useCountUp({ redraw: true }) isn't triggering a re-start on component re-render

### ISSUE THREAD

**Issue by omega-station** [glennreyes/react-countup#307]
Perhaps I'm doing something wrong and can't get this to work. For context:

```js
import React from 'react';
import { useCountUp } from 'react-countup';
import defaults from './defaults';
import { Props } from './definition';
import StyledChartCircle from './style';

const ChartCircle = (props: Props) => {
  const { isWaypoint, index, label, percent }: Props = { ...defaults, ...props };
  const animation: { [key: string]: number } = {
    index,
    delay: index * 0.275,
    duration: 1.5,
    percent: 3,
  };

  // @ts-ignore
  const { countUp } = useCountUp({ delay: animation.delay, duration: animation.duration, end: percent, start: 0, redraw: true, useEasing: true });

  return (
    <StyledChartCircle animation={animation} isWaypoint={isWaypoint} percent={percent}>
      <svg viewBox="0 0 36 36" aria-hidden="true">
        <path d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831" />
        <path d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831" />
        {isWaypoint && <text>{countUp}%</text>}
      </svg>
      <span>{label}</span>
    </StyledChartCircle>
  );
};

export default ChartCircle;
```

State var is `isWaypoint`; SVG animation is working as intended (reseting and replaying) when state is updated. However, `countUp` doesn't reset and start again with `redraw` set to `true`. Am I missing something?

PS ~ `redraw` and `preserveValue` (and perhaps other props) need to be added to the `useCountUp` TypeScript definition... I've needed to add @ts-ignore in order to use these.

**Comment by mmarkelov** [comment]
@omega-station there is no support for **redraw** and **preserveValue** options for ** useCountUp**, it works only for **CountUp** component, but for me it seems like that it is bug with it, cause there is check for  **redraw** in **shouldComponentUpdate**, and there is no checks in **componentDidUpdate**:
https://github.com/glennreyes/react-countup/blob/a3c03dd29583d0a9aa937f109db8c73d5ea9aea2/src/CountUp.js#L104-L116

So:
- I suppose this should be fixed for **CountUp** component
- **redraw** and **preserveValue** options should be add for ** useCountUp**

**Comment by omega-station** [comment]
@mmarkelov Thank you for your response--I agree with your recommendations 😺

**Comment by mmarkelov** [comment]
@omega-station #321 is ready for review, so **redraw** logic will be fixed, also I suppose  `preserveValue` doesn't needed, cause it can be made with `update` function:
```js
import React from 'react';
import {useCountUp} from "react-countup";

function App() {
  const { countUp, update } =  useCountUp({ duration: 2, end: 10000 });

  return (
    <div className="App">
      <header className="App-header">
        <text>{countUp}</text>
      </header>
    </div>
  );
}

export default App;
```

**Comment by mmarkelov** [comment]
should be fixed in **react-countup@5.0.0-0**
